test/unittests/t_telemetry_aggregator.cc

```diff
@@ -0,0 +1,266 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include "gtest/gtest.h"
+
+#include <cstdlib>
+
+#include "options.h"
+#include "util/file_guard.h"
+#include "util/posix.h"
+#include "util/string.h"
+
+#include "statistics.h"
+#include "telemetry_aggregator.h"
+#include "telemetry_aggregator_influx.h"
+
+using namespace std;  // NOLINT
+
+namespace perf {
+
+class T_TelemetryAggregator : public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    FILE *temp_file = CreateTempFile("./cvmfs_ut_TelemetryAggregator", 0600,
+                                     "w", &config_file_);
+    ASSERT_TRUE(temp_file != NULL);
+    unlink_guard_.Set(config_file_);
+
+    bool success = SafeWriteToFile("CVMFS_TELEMETRY_SEND=ON\n"
+            "CVMFS_TELEMETRY_RATE=5\n"
+            "CVMFS_INFLUX_HOST=localhost\n"
+            "CVMFS_INFLUX_PORT=8092\n"
+            "CVMFS_INFLUX_METRIC_NAME=influx_test",
+            config_file_,
+            0600);
+    ASSERT_TRUE(success);
+    int result = fclose(temp_file);
+    ASSERT_EQ(0, result);
+
+    c1_ = statistics_.Register("test.c1", "test counter 1");
+    ASSERT_TRUE(c1_ != NULL);
+    c2_ = statistics_.Register("test.c2", "test counter 2");
+    ASSERT_TRUE(c2_ != NULL);
+    c3_ = statistics_.Register("test.c3", "test counter 3");
+    ASSERT_TRUE(c3_ != NULL);
+
+    options_manager_.ParsePath(config_file_, false);
```

Comment:
For this PR the approach is fine.  In general, we could have used a series `options_manager_->SetValue()` calls and avoid writing a temporary file.